definitions/npm/ramda_v0.26.x/flow_v0.104.x-/test_ramda_v0.26.x_list.js

```diff
@@ -822,4 +818,26 @@ const str: string = "hello world";
     s: a,
     y: b
   }))([1, 2, 3])(["1", "2", "3"]);
+
+  describe('nth', () => {
+    it('should pass', () => {
+      const nthxs1: ?number = nth(2, [1,2,3]);
+      const nthxs2: ?number = nth(2)([1,2]);
+      const nthxs3: ?string = nth(2, ["curry"]);
+      const nthxs4: ?string = nth(2)(["curry"]);
+      const nthxs5: string = nth(2)("curry");
+      const nthxs6: string = nth(2, "curry");
+    });
+
+    it('should fail', () => {
+      //$ExpectError
+      const nthxs: ?string = nth(2)([1, 2, 3]);
+      //$ExpectError
+      const nthxs1: string = nth(2, [1, 2, 3]);
+      //$ExpectError
+      const nthxs2: ?number = nth(2)(['foo']);
+      //$ExpectError
+      const nthxs3: string = nth(2)([1, 2, 3]);
+    });
+  })
```

Comment:
I would add additional cases for the following scenarios:
* Passing a non-number to the first argument (fail).
* Passing an object to the second argument (fail).
* Passing a `$ReadOnlyArray` to the second argument (pass).
* Passing an `Array` to the second argument (pass).
* Passing a Tuple to the second argument (pass).
  * For now let's assume a uniform tuple type. We can make a separate PR for covering getting individual tuple elements, which I think has a lot of value but beyond the scope here.

For the array variants called out in this comment, I would not rely on a direct literal to be passed in.

Reply:
Done, hopefully I didn't miss any cases you specified above, let me know if I did :)

Reply:
Everything looks good for these, except I don't see a Tuple test. Can you add that or point me to where I missed it?